ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/handler/TestAbortedTxnCleaner.java

```diff
@@ -282,9 +284,19 @@ public void testCleaningOfAbortedDirectoriesBelowBase() throws Exception {
     cleaner.setCleanupHandlers(Arrays.asList(mockedTaskHandler));
     cleaner.run();
 
-    Mockito.verify(mockedFSRemover, Mockito.times(1)).clean(any(CleanupRequest.class));
+    Mockito.verifyNoInteractions(mockedFSRemover);
     Mockito.verify(mockedTaskHandler, Mockito.times(1)).getTasks();
 
+    directories = getDirectories(conf, t, null);
+    // Both base and delta files are present since the cleaner skips them as there is a newer write.
+    Assert.assertEquals(5, directories.size());
+    Assert.assertEquals(1, directories.stream().filter(dir -> dir.getName().startsWith(AcidUtils.BASE_PREFIX)).count());
+
+    // Run compaction and clean up
+    startInitiator();
+    startWorker();
+    startCleaner();
+
```

Comment:
I am not sure why we are calling worker and cleaner again. The scope of this test is to remove committed and aborted deltas below the base (compacted) file in a cleaner the first time.

Reply:
Ack. Should be the sam reason as previous comment. Please add an assert to check whether 1 compaction record is indeed there.